### ISSUE TITLE
Trouble when I compile

### ISSUE THREAD

**Issue by tacnoman** [michalmuskala/mongodb_ecto#100]
Hello guys.

I'm trying to use this lib, but I'm receiving this error:

```bash
$ mix compile
==> mongodb_ecto
Compiling 12 files (.ex)
warning: undefined behaviour function stop/3 (for behaviour Ecto.Adapter)
  lib/mongo_ecto.ex:1

warning: the variable "pipeline" is unsafe as it has been set inside a case/cond/receive/if/&&/||. Please explicitly return the variable value instead. For example:

    case int do
      1 -> atom = :one
      2 -> atom = :two
    end

should be written as

    atom =
      case int do
        1 -> :one
        2 -> :two
      end

Unsafe variable found at:
  lib/mongo_ecto/normalized_query.ex:86


== Compilation error on file lib/mongo_ecto/admin_pool.ex ==
** (CompileError) lib/mongo_ecto/admin_pool.ex:4: module Mongo.Pool is not loaded and could not be found
    (elixir) expanding macro: Kernel.use/2
    lib/mongo_ecto/admin_pool.ex:4: Mongo.Ecto.AdminPool (module)
    (elixir) lib/kernel/parallel_compiler.ex:116: anonymous fn/4 in Kernel.ParallelCompiler.spawn_compilers/1

could not compile dependency :mongodb_ecto, "mix compile" failed. You can recompile this dependency with "mix deps.compile mongodb_ecto", update it with "mix deps.update mongodb_ecto" or clean it with "mix deps.clean mongodb_ecto"
```

There is an pre requisite to use?

This is my deps:
```
  defp deps do
    [{:phoenix, "~> 1.1.6"},
     {:phoenix_pubsub, "~> 1.0"},
     {:phoenix_ecto, "~> 2.0.2"},
     {:postgrex, ">= 0.0.0"},
     {:phoenix_html, "~> 2.6"},
     {:phoenix_live_reload, "~> 1.0", only: :dev},
     {:gettext, "~> 0.11"},
     {:cowboy, "~> 1.0"},
     {:mongodb_ecto, "~> 0.1"},
     {:ecto, "~> 1.1.9"}
    ]
  end
```

My application
```
  def application do
    [mod: {AppName, []},
     applications: [:phoenix, :phoenix_pubsub, :phoenix_html, :cowboy, :logger, :gettext,
                    :phoenix_ecto, :postgrex, :mongodb_ecto, :ecto]]
  end
```

Thanks for this.

**Comment by michalmuskala** [comment]
Duplicate of https://github.com/michalmuskala/mongodb_ecto/issues/99 and https://github.com/michalmuskala/mongodb_ecto/pull/98

**Comment by michalmuskala** [comment]
I've just released 0.1.5, which has a stricter dependency on mongodb, and should fix this issue.
